src/bytecode/instructions.rs

```diff
@@ -15,7 +16,8 @@ impl PartialEq for Value {
         match (self, other) {
             (Value::Nil, Value::Nil) => true,
             (Value::Boolean(l), Value::Boolean(r)) => l == r,
-            (Value::Number(l), Value::Number(r)) =>
+            (Value::Integer(l), Value::Integer(r)) => l == r,
+            (Value::Float(l), Value::Float(r)) =>
                 l.partial_cmp(r).unwrap() == Ordering::Equal,
             (Value::Str(l), Value::Str(r)) => l == r,
             (_, _) => false
```

Comment:
This means, I think, that `Integer(1) != Float(1)`

Reply:
That is true. At first I was thinking I should have a different method (maybe in a `logical_operators` module) which converts the values to the same type, and then compares them. I wanted the `eq` method to only compare values of the same type. I am not sure if this is a good solution or not.

Reply:
Realistically, you probably want to have a decentralised `eq` method on values.